### PR TITLE
[3/n] add additional rich version check

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Related #9647 

Checks the rich version to ensure its above the minimum + add logic as Rich does not provide a `__version__`. @ananthsub mentioned issues with previous rich versions hence the version check!

cc @kaushikb11 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by kaushikb11** [comment]
@SeanNaren Should we include a warning in this PR that the installed Rich version is not >=10.2.2?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/callbacks/rich_model_summary.py [hunk 1]
```diff
@@ -61,7 +61,8 @@ class RichModelSummary(ModelSummary):
     def __init__(self, max_depth: int = 1) -> None:
         if not _RICH_AVAILABLE:
             raise ImportError(
-                "`RichModelSummary` requires `rich` to be installed. Install it by running `pip install rich`."
+                "`RichModelSummary` requires `rich>=10.2.2` to be installed. "
```
Comment (kaushikb11) [inline pytorch_lightning/callbacks/rich_model_summary.py:64]: We have two separate conditions here. Hence, I think we should have two different errors being raised.
One for rich not being installed, other for Rich not being >=10.2.2

Comment (SeanNaren) [reply pytorch_lightning/callbacks/rich_model_summary.py:64]: Will address!

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by ananthsub — APPROVED** [review]

**Pull request merged**